zypp/Digest.cc

```diff
@@ -137,13 +143,8 @@ namespace zypp {
     {
       if(initialized)
       {
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
-        EVP_MD_CTX_cleanup(mdctx);
-#else
-        EVP_MD_CTX_free(mdctx);
-#endif
-    	initialized = false;
-    	finalized = false;
+        initialized = false;
```

Comment:
I leave it up to you. Personally I'd eliminate `bool initialized` in `Digest::P`.

All tests for `initialized` can be replaced by tests for `mdctx`. Creation of the context in the `if(!initialized)` in `maybeInit` could use a local `EvpDataPtr` which is assigned to `mdctx` once all init steps succeeded (replaces `initialized = true;`). Here in `cleanup` the `initialized = false;` would be replaced by `mdctx.reset();` to invalidate the context.

Reply:
Done